modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationConfiguration.java

```diff
@@ -0,0 +1,549 @@
+    }
+
+    /**
+     * @return Number of received messages after which acknowledgment is sent.
+     */
+    public int ackSendThreshold() {
+        return ackSndThreshold;
+    }
+
+    /**
+     * @param ackSndThreshold New number of received messages after which acknowledgment is sent.
+     */
+    public void ackSendThreshold(int ackSndThreshold) {
+        this.ackSndThreshold = ackSndThreshold;
+    }
+
+    /**
+     * @return Maximum number of unacknowledged messages.
+     */
+    public int unackedMsgsBufferSize() {
+        return unackedMsgsBufSize;
+    }
+
+    /**
+     * @param unackedMsgsBufSize New maximum number of unacknowledged messages.
+     */
+    public void unackedMsgsBufferSize(int unackedMsgsBufSize) {
+        this.unackedMsgsBufSize = unackedMsgsBufSize;
+    }
+
+    /**
+     * @return Socket write timeout.
+     */
+    public long socketWriteTimeout() {
+        return sockWriteTimeout;
+    }
+
+    /**
+     * @param sockWriteTimeout New socket write timeout.
+     */
+    public void socketWriteTimeout(long sockWriteTimeout) {
+        this.sockWriteTimeout = sockWriteTimeout;
+    }
+
+    /**
+     * @return Bound port.
+     */
+    public int boundTcpPort() {
+        return boundTcpPort;
+    }
+
+    /**
+     * @param boundTcpPort New bound port.
+     */
+    public void boundTcpPort(int boundTcpPort) {
+        this.boundTcpPort = boundTcpPort;
+    }
+
+    /**
+     * @return Bound port for shared memory server.
+     */
+    public int boundTcpShmemPort() {
+        return boundTcpShmemPort;
+    }
+
+    /**
+     * @param boundTcpShmemPort New bound port for shared memory server.
+     */
+    public void boundTcpShmemPort(int boundTcpShmemPort) {
+        this.boundTcpShmemPort = boundTcpShmemPort;
+    }
+
+    /**
+     * @return Count of selectors to use in TCP server.
+     */
+    public int selectorsCount() {
+        return selectorsCnt;
+    }
+
+    /**
+     * @param selectorsCnt New count of selectors to use in TCP server.
+     */
+    public void selectorsCount(int selectorsCnt) {
+        this.selectorsCnt = selectorsCnt;
+    }
+
+    /**
+     * @return Complex variable that represents this node IP address.
+     */
+    public InetAddress localHost() {
+        return locHost;
+    }
+
+    /**
+     * @param locHost New complex variable that represents this node IP address.
+     */
+    public void localHost(InetAddress locHost) {
+        this.locHost = locHost;
+    }
+
+    /**
+     * @return Defines how many non-blocking  should be made before falling into  in NIO server. Long value. Default is
+     * . Can be set to  so selector threads will never block.
+     */
+    public long selectorSpins() {
+        return selectorSpins;
+    }
+
+    /**
+     * @param selectorSpins New defines how many non-blocking  should be made before falling into  in NIO server. Long
+     * value. Default is . Can be set to  so selector threads will never block.
```

Comment:
Confusing javadoc.

Reply:
IMHO javadoc on 'private' is for developer and on 'public' method is for user.
Javadoc here should contains very simple explanation and defaults.
May be it make sense to note that it is advanced option and there is nothing to change usually.